### PR TITLE
[COGS-1654] Setup cogs cli

### PR DESCRIPTION
**Pull request by mkatychev**
* implemented cli
* refactored cog schema

### CODE REVIEW

File: cmd/cogs/main.go [hunk 1]
```diff
@@ -2,8 +2,29 @@ package main
 
 import (
 	"fmt"
+	"os"
+
+	"github.com/bestowinc/cogs"
```
Comment (caquillo07) [inline cmd/cogs/main.go:7]: not that it matters, but just curious here. I noticed the actual URL for our github is capitalized on Bestowinc, do we favor one or the other or just  🤷‍♂️

File: cmd/cogs/main.go [hunk 2]
```diff
@@ -2,8 +2,29 @@ package main
...
+
+	opts, _ := docopt.ParseArgs(usage, os.Args[1:], "0.1")
+	var conf struct {
+		Generate bool
+		Env      string
+		File     string `docopt:"<cog-file>"`
+	}
+
+	opts.Bind(&conf)
+	_ = fmt.Println
```
Comment (caquillo07) [inline cmd/cogs/main.go:23]: curious on the reasoning of this println

File: generate.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	Name string
+	Envs map[string]Env
+}
+
+type Env struct {
+	Name string
+	Cfgs []Cfg
+}
+
+type Cfg struct {
```
Comment (caquillo07) [inline generate.go:18]: all the fields within this struct are private, does this need to be public?

File: cmd/cogs/main.go (hunk 1 above)
Comment (mkatychev) [reply cmd/cogs/main.go:7]: what's your preference? I'll go with that.

File: cmd/cogs/main.go (hunk 2 above)
Comment (mkatychev) [reply cmd/cogs/main.go:23]: whoops, forgot to leave it out.

File: generate.go (hunk 3 above)
Comment (mkatychev) [reply generate.go:18]: It does not, I find when starting from scratch making most fields public makes it easier to iterate and making them private later, perhaps it's a bad habit though.

File: cmd/cogs/main.go (hunk 1 above)
Comment (caquillo07) [reply cmd/cogs/main.go:7]: The actual address to our repo lol [https://github.com/Bestowinc/cogs/pull/1](https://github.com/Bestowinc/cogs) personally I prefer how you have it here, but as long as we consistent Idc really

File: some-service.cog.toml [hunk 4]
```diff
@@ -1,44 +1,54 @@
...
-[[docker]]
-
-[[docker.cfg]]
-[docker.cfg.var]
-value = 'var_value'
-
-[[docker.cfg]]
-[docker.cfg.other_var]
-value = 'other_var_value'
+name = ["some_service"]
```
Comment (mkatychev) [inline some-service.cog.toml:1]: whoops, need to revert this.
```suggestion
name = "some_service"
```

File: cmd/cogs/main.go (hunk 2 above)
Comment (mkatychev) [inline cmd/cogs/main.go:23]:
```suggestion
```

File: generate.go (hunk 3 above)
Comment (caquillo07) [reply generate.go:18]: I noticed its used on PR #2, so just to record it here, ignoring this file for this review.

**Pull request merged**